### PR TITLE
Update error_prone_annotations to 2.3.4

### PR DESCRIPTION
**Pull request by elharo**
## Description
Add dependency_management to update old dependency pulled in via cassandra project

## Motivation and Context
Unblocking Guava upgrade

## Impact
none

## Test Plan
CI

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-cassandra/pom.xml [hunk 1]
```diff
@@ -14,6 +14,16 @@
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
     </properties>
+    
+    <dependencyManagement>
```
Comment (tdcmeehan) [inline presto-cassandra/pom.xml:18]: Formatting looks off--looks like indented with tabs instead of spaces?

Comment (elharo) [reply presto-cassandra/pom.xml:18]: should be fixed now

**Review by tdcmeehan — APPROVED** [review]

**Comment by steveburnett** [comment]
Consider adding a release note entry, like 

```
== RELEASE NOTES ==

Security Changes
* Upgrade error_prone_annotations to 2.3.4 :pr:`22583`
```

**Comment by tdcmeehan** [comment]
@steveburnett I don't think that's necessary, since it's a build related upgrade and doesn't have any user-facing functionality differences.

**Comment by elharo** [comment]
Agreed. This is not itself a security fix. It's just one step along the way to upgrading the thing we actually care about.

**Pull request merged**